test/unit/SourceLocator/LocatedSourceTest.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace BetterReflectionTest\SourceLocator;
+
+use BetterReflection\SourceLocator\LocatedSource;
+use InvalidArgumentException;
+
+/**
+ * @covers \BetterReflection\SourceLocator\LocatedSource
+ */
+class LocatedSourceTest extends \PHPUnit_Framework_TestCase
+{
+    public function testValuesHappyPath()
+    {
+        $source = '<?php echo "Hello world";';
+        $file = 'path/to/file.php';
+        $locatedSource = new LocatedSource($source, $file);
+
+        $this->assertSame($source, $locatedSource->getSource());
+        $this->assertSame($file, $locatedSource->getFileName());
+    }
+
+    public function testValuesWithNullFilename()
+    {
+        $source = '<?php echo "Hello world";';
+        $file = null;
+        $locatedSource = new LocatedSource($source, $file);
```

Comment:
Noooooo it does not get `eval()`'d, it gets loaded directly into the AST (rather than being loaded from a file)